### ISSUE TITLE
Gateway fails to pay despite LND paying invoice just fine

### ISSUE THREAD

**Issue by TonyGiorgio** [fedimint/fedimint#3990]
I can't get a single payment out through the gateway on mainnet but when I just use `payinvoice` normally, LND doesn't have a problem. Running 0.2.1-rc3 for both fedimintd & gatewayd. 

Invoice
```
lnbc500n1pjcve32pp50vwalaqqahfmntcgnngp4dgr4hwfpwv0wspaup6elpnlfa3ddujqdqqcqzzsxqrrsssp5fyr8g355fdd9wcl8wl6e3nm9khyg5khq0wfqqkhz9q5xd2nsw78q9qyyssq74fvwpz3a4v7qg7v9mlwuwnqefyu9ertz54y0q82ntp06mm7wwekrh5znx2fgg6vmqpx74u4q807wvxeyh6p4zzuwa8lftdcvqje6wcq5hpmle
```

Gateway: 
```
docker-fedimintd-1    | 2023-12-23T04:46:59.036752Z  INFO consensus: Session 371 completed                                                                                                                
docker-gatewayd-1     | 2023-12-23T04:47:12.239810Z  INFO ln_gateway::lnd: LND Paying invoice PrunedInvoice { amount: Amount { msats: 50000 }, destination: PublicKey(028fe7cb1b6cd6bac7f5c9c4eff166fdb73a
b861595e5b7f69b099bb32091dccede5828671f691268fc4cf1937989d30aa0a6213c0117103d3e7b553455cf313), destination_features: [2, 66, 0], payment_hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62
d6f24, payment_secret: [73, 6, 116, 70, 148, 75, 90, 87, 99, 231, 119, 245, 152, 207, 101, 181, 200, 138, 90, 224, 123, 146, 0, 90, 226, 40, 40, 102, 170, 112, 119, 142], route_hints: [], min_final_cltv
_delta: 80, expiry_timestamp: 1703310394 }                                                                                                                                                                
docker-lnd-1          | 2023-12-23 04:47:12.286 [ERR] RPCS: [/routerrpc.Router/TrackPaymentV2]: rpc error: code = NotFound desc = payment isn't initiated                                                 
docker-lnd-1          | 2023-12-23 04:47:12.381 [WRN] CRTR: Channel 832434755940515841 has zero cltv delta                                                                                                
docker-lnd-1          | 2023-12-23 04:47:12.437 [WRN] CRTR: Failed to find route for payment 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24: unable to find a path to destination       
docker-lnd-1          | 2023-12-23 04:47:12.448 [ERR] CRTR: Payment 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24 failed: no_route                                                     
docker-gatewayd-1     | 2023-12-23T04:47:12.449525Z  INFO ln_gateway::lnd: LND payment failed for invoice PrunedInvoice { amount: Amount { msats: 50000 }, destination: PublicKey(028fe7cb1b6cd6bac7f5c9c4
eff166fdb73ab861595e5b7f69b099bb32091dccede5828671f691268fc4cf1937989d30aa0a6213c0117103d3e7b553455cf313), destination_features: [2, 66, 0], payment_hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0
759f867f4f62d6f24, payment_secret: [73, 6, 116, 70, 148, 75, 90, 87, 99, 231, 119, 245, 152, 207, 101, 181, 200, 138, 90, 224, 123, 146, 0, 90, 226, 40, 40, 102, 170, 112, 119, 142], route_hints: [], mi
n_final_cltv_delta: 80, expiry_timestamp: 1703310394 } with Payment { payment_hash: "7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24", value: 50, creation_date: 1703306832, fee: 0, paym
ent_preimage: "0000000000000000000000000000000000000000000000000000000000000000", value_sat: 50, value_msat: 50000, payment_request: "", status: Failed, fee_sat: 0, fee_msat: 0, creation_time_ns: 170330
6832287725893, htlcs: [], payment_index: 19, failure_reason: FailureReasonNoRoute }                                                                                                                       
docker-gatewayd-1     | 2023-12-23T04:47:12.449610Z  WARN ln_gateway::state_machine::pay: Failed to buy preimage with Payment failed: FailureReasonNoRoute for contract OutgoingContractAccount { amount: 
Amount { msats: 50051 }, contract: OutgoingContract { hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24, gateway_key: PublicKey(b81ca3e6ea9ed786736fa265a923ac7af71d53b6b48a40d06c759
3433f4cf18d14dc2ce8d12e3771905629cb12cb436f9f864b3f569f5b5f4ba905a3f41cf737), timelock: 823014, user_key: PublicKey(269ef76e6a62ab2776d2c578939962c4e0d03df7d75a22686bb1326674a66ae7464937e2acd81ff8fac555
9f9d130892ae7c5e2ece0f906034afedd440409e85), cancelled: false } }                                                                                                                                         
docker-gatewayd-1     | 2023-12-23T04:47:12.449721Z  INFO state_machine_transition{operation_id=6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761}: fedimint_client::sm::executor: Executin
g state transition                                                                                                                                                                                        
docker-gatewayd-1     | 2023-12-23T04:47:12.450541Z  INFO state_machine_transition{operation_id=6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761}: fedimint_client::sm::executor: Executin
g state transition                                                                                                                                                                                        
docker-gatewayd-1     | 2023-12-23T04:47:12.450600Z  INFO state_machine_transition{operation_id=6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761}: ln_gateway::state_machine::pay: Canceling outgoing contract OutgoingContractAccount { amount: Amount { msats: 50051 }, contract: OutgoingContract { hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24, gateway_key: PublicKey(b81ca3e6ea9ed786736fa265a923ac7af71d53b6b48a40d06c7593433f4cf18d14dc2ce8d12e3771905629cb12cb436f9f864b3f569f5b5f4ba905a3f41cf737), timelock: 823014, user_key: PublicKey(269ef76e6a62ab2776d2c578939962c4e0d03df7d75a22686bb1326674a66ae7464937e2acd81ff8fac5559f9d130892ae7c5e2ece0f906034afedd440409e85), cancelled: false } }
docker-gatewayd-1     | 2023-12-23T04:47:12.450916Z  INFO state_machine_transition{operation_id=6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761}: ln_gateway::state_machine::pay: Canceled outgoing contract OutgoingContractAccount { amount: Amount { msats: 50051 }, contract: OutgoingContract { hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24, gateway_key: PublicKey(b81ca3e6ea9ed786736fa265a923ac7af71d53b6b48a40d06c7593433f4cf18d14dc2ce8d12e3771905629cb12cb436f9f864b3f569f5b5f4ba905a3f41cf737), timelock: 823014, user_key: PublicKey(269ef76e6a62ab2776d2c578939962c4e0d03df7d75a22686bb1326674a66ae7464937e2acd81ff8fac5559f9d130892ae7c5e2ece0f906034afedd440409e85), cancelled: false } } with txid c8f25df4cf81475724b154dda9ef261a5be1d27d05f35ddc0e584179144cbe6b
docker-gatewayd-1     | 2023-12-23T04:47:12.451181Z  INFO pay_invoice: ln_gateway::state_machine: Got state CancelContract(GatewayPayCancelContract { contract: OutgoingContractAccount { amount: Amount { msats: 50051 }, contract: OutgoingContract { hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24, gateway_key: PublicKey(b81ca3e6ea9ed786736fa265a923ac7af71d53b6b48a40d06c7593433f4cf18d14dc2ce8d12e3771905629cb12cb436f9f864b3f569f5b5f4ba905a3f41cf737), timelock: 823014, user_key: PublicKey(269ef76e6a62ab2776d2c578939962c4e0d03df7d75a22686bb1326674a66ae7464937e2acd81ff8fac5559f9d130892ae7c5e2ece0f906034afedd440409e85), cancelled: false } }, error: OutgoingPaymentError { error_type: LightningPayError { lightning_error: FailedPayment { failure_reason: "FailureReasonNoRoute" } }, contract_id: 6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761, contract: Some(OutgoingContractAccount { amount: Amount { msats: 50051 }, contract: OutgoingContract { hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24, gateway_key: PublicKey(b81ca3e6ea9ed786736fa265a923ac7af71d53b6b48a40d06c7593433f4cf18d14dc2ce8d12e3771905629cb12cb436f9f864b3f569f5b5f4ba905a3f41cf737), timelock: 823014, user_key: PublicKey(269ef76e6a62ab2776d2c578939962c4e0d03df7d75a22686bb1326674a66ae7464937e2acd81ff8fac5559f9d130892ae7c5e2ece0f906034afedd440409e85), cancelled: false } }) } }) while awaiting for output of 6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761
docker-gatewayd-1     | 2023-12-23T04:47:13.461044Z  INFO state_machine_transition{operation_id=6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761}: fedimint_client::sm::executor: Executing state transition
docker-gatewayd-1     | 2023-12-23T04:47:13.461878Z ERROR pay_invoice: ln_gateway: Cancelled with OutgoingContractError: An error occurred while paying the lightning invoice. while paying invoice: 6348f23823871e0a73e4961fec0ca5f084aa2c68001330f11cb0e43020065761
docker-gatewayd-1     | 2023-12-23T04:47:13.461993Z ERROR pay_invoice: ln_gateway::rpc::rpc_server: error=Outgoing Payment Error OutgoingContractError: An error occurred while paying the lightning invoice.
```

Same LND node right after:
```
/ # lncli payinvoice lnbc500n1pjcve32pp50vwalaqqahfmntcgnngp4dgr4hwfpwv0wspaup6elpnlfa3ddujqdqqcqzzsxqrrsssp5fyr8g355fdd9wcl8wl6e3nm9khyg5khq0wfqqkhz9q5xd2nsw78q9qyyssq74fvwpz3a4v7qg7v9mlwuwnqefyu9ertz5
4y0q82ntp06mm7wwekrh5znx2fgg6vmqpx74u4q807wvxeyh6p4zzuwa8lftdcvqje6wcq5hpmle
Payment hash: 7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24
Description: 
Amount (in satoshis): 50
Fee limit (in satoshis): 50
Destination: 03cc1d0932bb99b0697f5b5e5961b83ab7fd66f1efc4c9f5c7bad66c1bcbe78f02
Confirm payment (yes/no): yes
+------------+--------------+--------------+--------------+-----+----------+--------------------+--------------------------------------------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT           | ROUTE                                      |
+------------+--------------+--------------+--------------+-----+----------+--------------------+--------------------------------------------+
| SUCCEEDED  |        0.074 |        2.848 | 50           | 1   |   822717 | 904307631956361217 | Voltage-C2->No🥈Best->⚡️ STACKER.NEWS ⚡️ |
+------------+--------------+--------------+--------------+-----+----------+--------------------+--------------------------------------------+
Amount + fee:   50 + 1 sat
Payment hash:   7b1ddff400edd3b9af089cd01ab503addc90b98f7403de0759f867f4f62d6f24
Payment status: SUCCEEDED
```

I've done this several times with different destinations, no success. I can send to another user's invoice inside the federation and I can receive via lightning. It's just that something is causing issues using the new `SendPaymentV2` API. I was not seeing this on signet so I'm a bit confused what could be the problem here. 

Could it be something fee related? I'm not sure how fedimint is setting max fee when making the LND request.

**Comment by TonyGiorgio** [comment]
Okay this was user error. I originally set up the gateway with a `1` *m*sat base fee and low ppm but I had meant 1 sat fee. After increasing this to 1000 msat, the payment went through. 

I think I noticed I also had to leave federation and add it again after a config change for it to reflect in the federation. A potential improvement could be made on the gatewayd to make routing fee changes instant. I wasn't sure if leaving and adding again would keep my balance and the `backup` command is still unimplemented. So a bit weird changing fees right now.

**Comment by m1sterc001guy** [comment]
> I think I noticed I also had to leave federation and add it again after a config change for it to reflect in the federation

Made an issue for [this](I think I noticed I also had to leave federation and add it again after a config change for it to reflect in the federation). I believe the config change would have updated if you waited until the next time the gateway registered, which should have been within 10 minutes. This is a long time to wait though and there's no reason to wait, we should just trigger registration after making a config change.
